lib/internal/modules/esm/get_format.js

```diff
@@ -61,6 +65,8 @@ function defaultGetFormat(url, context, defaultGetFormatUnused) {
           'The Node.js specifier resolution in ESM is experimental.',
           'ExperimentalWarning');
         format = legacyExtensionFormatMap[ext];
+      } else if (statSync(parsed.pathname).isDirectory()) {
+        throw new ERR_UNSUPPORTED_DIR_IMPORT();
```

Comment:
I would prefer if we can just use the standard `ERR_MODULE_NOT_FOUND` for this.

I've also just landed https://github.com/nodejs/node/pull/31906 which has a nice new hint output for these sorts of cases. If we could get this case to just use the same output as that showing a `Did you mean to import ....` then that should avoid duplicate work between all these cases of mistakes.

Reply:
I tried to implement `ERR_MODULE_NOT_FOUND ` instead of creating a new error code. I don't think I have access to the `base` path. Maybe that's something that needs to be added to the `context` parameter?
Currently it prints (I haven't rebased to linked PR yet):

```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'file:///.../node/test/parallel/' imported from [object Object]
```

Reply:
After rebasing, it doesn't print the hint either, because it's triggered only by the resolve hook, and my changes are to the getFormat hook...

Reply:
Right, let's make sure the same hint code runs in this code path then.